java-frontend/src/test/java/org/sonar/java/se/Expectations.java

```diff
@@ -235,109 +230,191 @@ public void visitTrivia(SyntaxTrivia syntaxTrivia) {
       collectExpectedIssues(syntaxTrivia.comment(), syntaxTrivia.startLine());
     }
 
-    private void collectExpectedIssues(String comment, int line) {
-      if (comment.startsWith(NONCOMPLIANT_COMMENT)) {
-        parseIssue(comment, line);
+    @VisibleForTesting
+    void collectExpectedIssues(String comment, int line) {
+      if (NONCOMPLIANT_COMMENT.matcher(comment).find()) {
+        IssueAndFlows issueAndFlows = parseIssue(comment, line);
+        if (issueAndFlows.parsed()) {
```

Comment:
Given : 
- how the constant `IssueAndFlow.NULL` is declared and used
- IssueAndFlow class is not implementing equals and hashcode

I don't see how this condition can be false and thus I don't see how it is tested.
I would go for the suggestion to remove it completely.